sqlmesh/core/loader.py

```diff
@@ -54,42 +53,53 @@ class LoadedProject:
     standalone_audits: UniqueKeyDict[str, StandaloneAudit]
     audits: UniqueKeyDict[str, ModelAudit]
     metrics: UniqueKeyDict[str, Metric]
-    dag: DAG[str]
     requirements: t.Dict[str, str]
     excluded_requirements: t.Set[str]
 
 
 class Loader(abc.ABC):
     """Abstract base class to load macros and models for a context"""
 
-    def __init__(self) -> None:
+    def __init__(self, context: GenericContext, path: Path) -> None:
         self._path_mtimes: t.Dict[Path, float] = {}
-        self._dag: DAG[str] = DAG()
+        self.context = context
+        self.context_path = path
```

Comment:
IMHO `context_path` is confusing, since there's one context but multiple paths. Shouldn't this be `project_path` or `config_path`?